Content.Server/GameTicking/GameTicker.Replays.cs

```diff
@@ -108,6 +112,37 @@ private void ReplaysOnRecordingFinished(ReplayRecordingFinished data)
         data.Directory.Rename(data.Path, state.MoveToPath.Value);
     }
 
+    private void ReplaysOnRecordingStopped(MappingDataNode metadata)
+    {
+        // Write round info like map and round end summery into the replay_final.yml file. Useful for external parsers.
+
+        metadata["map"] = new ValueDataNode(_gameMapManager.GetSelectedMap()?.MapName);
+        metadata["gamemode"] = new ValueDataNode(CurrentPreset != null ? Loc.GetString(CurrentPreset.ModeTitle) : string.Empty);
+
+        // These should be set to null to prepare them for the next round.
+        if (_replayRoundPlayerInfo != null)
+        {
+            string finalPlayers = SerializeToYaml(_replayRoundPlayerInfo);
+            metadata["roundEndPlayers"] = new ValueDataNode(finalPlayers);
+            metadata["roundEndText"] = new ValueDataNode(_replayRoundText);
+            metadata["server_id"] = new ValueDataNode(_configurationManager.GetCVar(CCVars.ServerId));
+            _replayRoundPlayerInfo = null;
+            _replayRoundText = null;
+        }
+        else
+        {
+            // If theres no players then there probably nothing interesting in round end text worth saving.
+            metadata["roundEndPlayers"] = new ValueDataNode(string.Empty);
+            metadata["roundEndText"] = new ValueDataNode(string.Empty);
+        }
+    }
+
+    private static string SerializeToYaml<T>(T data)
+    {
+        var serializer = new SerializerBuilder().Build();
+        return serializer.Serialize(data!);
+    }
```

Comment:
IMO it'd be better use serialization manager, and either turn RoundEndPlayerInfo into a `DataDefinition` with `DataField`s or use some custom type serializer